### PR TITLE
Remove `Array<Self>` from parsers

### PR DESCRIPTION
**Pull request by carson-katri**
Unfortunately, I still have no idea *why* this crashes the compiler (and only in release mode), but removing `Array<Self>` from the parsers seems to fix it.

I have also updated the `_GeneratedModifiers.swift` for the latest Xcode 15.1, changed `xrOS` to `visionOS` across the project, and replaced `Package.swift` with `Package@swift-5.9.swift` as 5.9 is the only supported Swift version.

Swift bug report: https://github.com/apple/swift/issues/70410

### CODE REVIEW

File: Sources/LiveViewNative/Stylesheets/Modifiers/Overrides/SearchScopesModifier.swift [hunk 1]
```diff
@@ -22,7 +22,7 @@ struct _SearchScopesModifier<R: RootRegistry>: ViewModifier {
     @ObservedElement private var element
     @LiveContext<R> private var context
     
-    #if os(iOS) || os(macOS) || os(tvOS) || os(xrOS)
+    #if os(iOS) || os(macOS) || os(tvOS) || os(visionOS)
```
Comment (simlay) [inline Sources/LiveViewNative/Stylesheets/Modifiers/Overrides/SearchScopesModifier.swift:25]: This is a more reasonable OS name.

File: Package@swift-5.9.swift
Comment (simlay) [inline Package@swift-5.9.swift:1]: Cool.

**Review by simlay — APPROVED** [review]
I understand and approve. ~of everything but `Array<Self>.parse`.~ The part where `Array<Self>.parse` fixes a compiler build confuses me.

File: Sources/LiveViewNative/Stylesheets/Modifiers/Overrides/SearchScopesModifier.swift (hunk 1 above)
Comment (carson-katri) [reply Sources/LiveViewNative/Stylesheets/Modifiers/Overrides/SearchScopesModifier.swift:25]: That was true of earlier betas, but Xcode now warns about `xrOS` not being a known platform. `visionOS` is the correct name.

**Pull request merged**